### PR TITLE
model loading

### PR DESCRIPTION
**Pull request by fschlatt**
This PR attempts to cleanup the model loading functionality.

1. Removes `BertConfig` and `ModernBertConfig` and just keeps the `BertBlocksConfig`
2. Adds `from_model` (convert an in-memory model into a bertblocks model) and `from_huggingface` (convert a pre-trained checkpoint into a bertblocks model) functions

### CODE REVIEW

File: bertblocks/integration/load_modernbert.py [hunk 1]
```diff
@@ -1,111 +1,104 @@
...
-                    bertblocks_model.encd.blocks[i].pre_norm_attn.bias.data.copy_(
-                        orig_model.layers[i].attn_norm.bias.data
-                    )
-
-            bertblocks_model.encd.blocks[i].pre_norm_ffwd.weight.data.copy_(orig_model.layers[i].mlp_norm.weight.data)
-            if bertblocks_config.norm_bias:
-                bertblocks_model.encd.blocks[i].pre_norm_ffwd.bias.data.copy_(orig_model.layers[i].mlp_norm.bias.data)
+        bertblocks_model = from_modernbert_model(
+            orig_model, add_pooling_layer=add_pooling_layer, attn_implementation=attn_implementation
+        )
```
[the remark this answers is not in the document]
Comment (fschlatt) [reply bertblocks/integration/load_modernbert.py]: Fixed: the initial `BertBlocksModel` allocation is now skipped when `load_weights=True` — the function returns directly from `from_modernbert_model`. https://github.com/coral-nlp/bertblocks/commit/4fcef603eb60b8deeb3b6f3f228a07a1d73ef310

File: bertblocks/integration/load_bert.py [hunk 2]
```diff
@@ -45,77 +51,108 @@ def from_bert_model(
...
+    )
     bertblocks_model = BertBlocksModel(bertblocks_config, add_pooling_layer=add_pooling_layer)
 
     if load_weights:
         orig_model = BertModel.from_pretrained(pretrained_model_name_or_path, add_pooling_layer=add_pooling_layer)
+        bertblocks_model = from_bert_model(
+            orig_model, add_pooling_layer=add_pooling_layer, attn_implementation=attn_implementation
+        )
+
+    return bertblocks_model
```
[the remark this answers is not in the document]
Comment (fschlatt) [reply bertblocks/integration/load_bert.py]: Fixed: the initial `BertBlocksModel` allocation is now skipped when `load_weights=True` — the function returns directly from `from_bert_model`. https://github.com/coral-nlp/bertblocks/commit/4fcef603eb60b8deeb3b6f3f228a07a1d73ef310

File: docs/configuration.md [hunk 3]
```diff
@@ -59,34 +59,36 @@ API reference for the full list.
...
 
 Reproduces the original BERT architecture:
 
 ```python
-from bertblocks.config import BertConfig
+from bertblocks.config import BertBlocksConfig
+from transformers import BertConfig
 
-config = BertConfig(vocab_size=30522)
+config = BertBlocksConfig.from_config(BertConfig())
```
[the remark this answers is not in the document]
Comment (fschlatt) [reply docs/configuration.md]: Fixed: updated the example to use `BertConfig.from_pretrained("bert-base-uncased")` so `mask_token_id` is populated correctly. https://github.com/coral-nlp/bertblocks/commit/4fcef603eb60b8deeb3b6f3f228a07a1d73ef310

File: docs/configuration.md [hunk 4]
```diff
@@ -59,34 +59,36 @@ API reference for the full list.
...
 
 Reproduces the ModernBERT architecture:
 
 ```python
-from bertblocks.config import ModernBertConfig
+from bertblocks.config import BertBlocksConfig
+from transformers import ModernBertConfig
 
-config = ModernBertConfig(vocab_size=50368)
+config = BertBlocksConfig.from_config(ModernBertConfig(vocab_size=50368))
```
[the remark this answers is not in the document]
Comment (fschlatt) [reply docs/configuration.md]: Fixed: updated the example to use `ModernBertConfig.from_pretrained("answerdotai/ModernBERT-base")` so `mask_token_id` is populated correctly. https://github.com/coral-nlp/bertblocks/commit/4fcef603eb60b8deeb3b6f3f228a07a1d73ef310

File: tests/integration/test_load_bert.py [hunk 5]
```diff
@@ -200,30 +200,33 @@ def test_attn(self, request, baseline_model_name, subtests, seq, hf_model, bb_mo
...
         with torch.no_grad():
             for layer_idx in range(len(hf_model.encoder.layer)):
                 with subtests.test(f"layer_{layer_idx}"):
                     torch.testing.assert_close(
-                        hf_model.encoder.layer[layer_idx](hf_emb, attention_mask=attention_mask)[0][
+                        hf_model.encoder.layer[layer_idx](hf_emb, attention_mask=hf_attention_mask)[
                             seq["attention_mask"].bool()
                         ],
-                        bb_model.encd.blocks[layer_idx](bb_emb, attention_mask=attention_mask)[0][
+                        bb_model.encd.blocks[layer_idx](bb_emb, attention_mask=bb_attention_mask)[0][
```
[the remark this answers is not in the document]
Comment (fschlatt) [reply tests/integration/test_load_bert.py:229]: Fixed: `hf_attention_mask` now uses `hf_model.config` instead of `bb_model.config`. Per your note, the boolean-mask indexing is intentional since transformers v5 BertLayer no longer returns a tuple. https://github.com/coral-nlp/bertblocks/commit/4fcef603eb60b8deeb3b6f3f228a07a1d73ef310

File: tests/integration/test_load_bert.py [hunk 6]
```diff
@@ -200,30 +200,33 @@ def test_attn(self, request, baseline_model_name, subtests, seq, hf_model, bb_mo
...
-                f"TestFromBertModel::test_attn[{baseline_model_name}]",
-            ],
-        )
+        # depends(
+        #     request,
+        #     [
+        #         f"TestFromBertModel::test_ffwd[{baseline_model_name}]",
+        #         f"TestFromBertModel::test_attn[{baseline_model_name}]",
+        #     ],
+        # )
```
[the remark this answers is not in the document]
Comment (fschlatt) [reply tests/integration/test_load_bert.py]: Fixed: re-enabled the `depends()` call in `test_attn` (line 178) and `test_blocks` (line 203). https://github.com/coral-nlp/bertblocks/commit/4fcef603eb60b8deeb3b6f3f228a07a1d73ef310

File: bertblocks/integration/load_modernbert.py [hunk 7]
```diff
@@ -1,111 +1,104 @@
...
-                bertblocks_model.norm.bias.data.copy_(orig_model.final_norm.bias.data)
+                bertblocks_model.encd.blocks[i].pre_norm_attn.bias.data.copy_(orig_model.layers[i].attn_norm.bias.data)
+
+        bertblocks_model.encd.blocks[i].pre_norm_ffwd.weight.data.copy_(orig_model.layers[i].mlp_norm.weight.data)
+        if bertblocks_config.norm_bias:
+            bertblocks_model.encd.blocks[i].pre_norm_ffwd.bias.data.copy_(orig_model.layers[i].mlp_norm.bias.data)
+
+        bertblocks_model.norm.weight.data.copy_(orig_model.final_norm.weight.data)
+        if bertblocks_config.norm_bias:
+            bertblocks_model.norm.bias.data.copy_(orig_model.final_norm.bias.data)
```
[the remark this answers is not in the document]
Comment (fschlatt) [reply bertblocks/integration/load_modernbert.py]: Fixed: moved the `bertblocks_model.norm` weight copy outside the per-layer loop so `final_norm` is transferred exactly once. https://github.com/coral-nlp/bertblocks/commit/4fcef603eb60b8deeb3b6f3f228a07a1d73ef310

File: bertblocks/integration/load_modernbert.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (fschlatt) [reply bertblocks/integration/load_modernbert.py]: Fixed: the initial `BertBlocksModel` allocation is now skipped when `load_weights=True` — the function returns directly from `from_modernbert_model`. https://github.com/coral-nlp/bertblocks/commit/4fcef603eb60b8deeb3b6f3f228a07a1d73ef310

File: bertblocks/integration/load_bert.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (fschlatt) [reply bertblocks/integration/load_bert.py]: Fixed: the initial `BertBlocksModel` allocation is now skipped when `load_weights=True` — the function returns directly from `from_bert_model`. https://github.com/coral-nlp/bertblocks/commit/4fcef603eb60b8deeb3b6f3f228a07a1d73ef310

File: docs/configuration.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (fschlatt) [reply docs/configuration.md]: Fixed: updated the example to use `BertConfig.from_pretrained("bert-base-uncased")` so `mask_token_id` is populated correctly. https://github.com/coral-nlp/bertblocks/commit/4fcef603eb60b8deeb3b6f3f228a07a1d73ef310

File: docs/configuration.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (fschlatt) [reply docs/configuration.md]: Fixed: updated the example to use `ModernBertConfig.from_pretrained("answerdotai/ModernBERT-base")` so `mask_token_id` is populated correctly. https://github.com/coral-nlp/bertblocks/commit/4fcef603eb60b8deeb3b6f3f228a07a1d73ef310

File: tests/integration/test_load_bert.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (fschlatt) [reply tests/integration/test_load_bert.py:229]: Fixed: `hf_attention_mask` now uses `hf_model.config` instead of `bb_model.config`. Per your note, the boolean-mask indexing is fine since transformers v5 BertLayer no longer returns a tuple. https://github.com/coral-nlp/bertblocks/commit/4fcef603eb60b8deeb3b6f3f228a07a1d73ef310

File: tests/integration/test_load_bert.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (fschlatt) [reply tests/integration/test_load_bert.py]: Fixed: re-enabled the `depends()` call in both `test_attn` and `test_blocks`. https://github.com/coral-nlp/bertblocks/commit/4fcef603eb60b8deeb3b6f3f228a07a1d73ef310

File: bertblocks/integration/load_modernbert.py (hunk 7 above)
[the remark this answers is not in the document]
Comment (fschlatt) [reply bertblocks/integration/load_modernbert.py]: Fixed: moved the `bertblocks_model.norm` weight copy outside the per-layer loop so `final_norm` is transferred exactly once. https://github.com/coral-nlp/bertblocks/commit/4fcef603eb60b8deeb3b6f3f228a07a1d73ef310

File: bertblocks/integration/load_modernbert.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (fschlatt) [reply bertblocks/integration/load_modernbert.py]: Fixed: the initial `BertBlocksModel` allocation is now skipped when `load_weights=True` — the function returns directly from `from_modernbert_model`. https://github.com/coral-nlp/bertblocks/commit/4fcef603eb60b8deeb3b6f3f228a07a1d73ef310

File: bertblocks/integration/load_bert.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (fschlatt) [reply bertblocks/integration/load_bert.py]: Fixed: the initial `BertBlocksModel` allocation is now skipped when `load_weights=True` — the function returns directly from `from_bert_model`. https://github.com/coral-nlp/bertblocks/commit/4fcef603eb60b8deeb3b6f3f228a07a1d73ef310

File: docs/configuration.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (fschlatt) [reply docs/configuration.md]: Fixed: updated the example to use `BertConfig.from_pretrained("bert-base-uncased")` so `mask_token_id` is populated correctly. https://github.com/coral-nlp/bertblocks/commit/4fcef603eb60b8deeb3b6f3f228a07a1d73ef310

File: docs/configuration.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (fschlatt) [reply docs/configuration.md]: Fixed: updated the example to use `ModernBertConfig.from_pretrained("answerdotai/ModernBERT-base")` so `mask_token_id` is populated correctly. https://github.com/coral-nlp/bertblocks/commit/4fcef603eb60b8deeb3b6f3f228a07a1d73ef310

File: tests/integration/test_load_bert.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (fschlatt) [reply tests/integration/test_load_bert.py:229]: Fixed: `hf_attention_mask` now uses `hf_model.config` instead of `bb_model.config`. Per your note, the boolean-mask indexing is fine since transformers v5 BertLayer no longer returns a tuple. https://github.com/coral-nlp/bertblocks/commit/4fcef603eb60b8deeb3b6f3f228a07a1d73ef310

File: tests/integration/test_load_bert.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (fschlatt) [reply tests/integration/test_load_bert.py]: Fixed: re-enabled the `depends()` call in both `test_attn` and `test_blocks`. https://github.com/coral-nlp/bertblocks/commit/4fcef603eb60b8deeb3b6f3f228a07a1d73ef310

File: bertblocks/integration/load_modernbert.py (hunk 7 above)
[the remark this answers is not in the document]
Comment (fschlatt) [reply bertblocks/integration/load_modernbert.py]: Fixed: moved the `bertblocks_model.norm` weight copy outside the per-layer loop so `final_norm` is transferred exactly once. https://github.com/coral-nlp/bertblocks/commit/4fcef603eb60b8deeb3b6f3f228a07a1d73ef310
